src/app/pages/property-details/property-details.component.ts

```diff
@@ -0,0 +1,96 @@
+import { Component, OnInit } from '@angular/core';
+import { PropertyDetailService } from '../../services/property-detail/property-detail.service';
+import { ActivatedRoute } from '@angular/router';
+import { NgxSpinnerService } from 'ngx-spinner';
+
+@Component({
+  selector: 'app-property-details',
+  templateUrl: './property-details.component.html',
+  styleUrls: ['./property-details.component.scss']
+})
+
+export class PropertyDetailsComponent implements OnInit {
+  property: Array<any> = [];
+  slug: string;
+  rooms: number;
+  bathrooms: number;
+  garages: number;
+  city: string;
+  state: string;
+  street: string;
+  title: string;
+  price: string;
+  description: string;
+  imageMain: string;
+  imageOthers: Array<any> = [];
+  lotSize: string;
+  video: string;
+  ifBuilding: boolean;
+  ifVideo: boolean;
+  createdAt: string;
+  clientName: string;
+  phone: string;
+  email: string;
+  clientStreet: string;
+  clientState: string;
+  clientCity: string;
+
+  constructor(
+    private route: ActivatedRoute,
+    private propertyservice: PropertyDetailService,
+    private spinner: NgxSpinnerService,
+  ) { }
+
+  ngOnInit(): void {
+    this.route.paramMap.subscribe(result => {
+      let slug = result.get('slug');
```

Comment:
Good work here. I was actually thinking if you could have thought of route resolvers but then i saw you implemented the spinner which is another approach to this, maybe you could think of this next time.